### PR TITLE
fix(terraform): support github repo source url with underscores

### PR DESCRIPTION
**Pull request by raspy8766**
## Changes:

Adds underscore to the list of whitelisted characters for the `githubRefMatchRegex` so that repositories with an underscore don't get truncated for Terraform and Terragrunt managers.
## Context:
Our repo has an underscore and was getting truncated like below which resulted in us getting the "Failed to look up dependency" warning message:
### Before
image

### After
image

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/manager/terraform/modules.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import type { PackageDependency } from '../types';
 import { extractTerraformProvider } from './providers';
 import { ExtractionResult, TerraformDependencyTypes } from './util';
 
-const githubRefMatchRegex = /github.com([/:])(?<project>[^/]+\/[a-z0-9-.]+).*\?ref=(?<tag>.*)$/;
+const githubRefMatchRegex = /github.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/;
```
Comment (viceice) [inline lib/manager/terraform/modules.ts:10]:
```suggestion
const githubRefMatchRegex = /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/i;
```
Match only dot and case insensitive.

File: lib/manager/terragrunt/modules.ts [hunk 2]
```diff
@@ -7,7 +7,7 @@ import type { PackageDependency } from '../types';
 import { extractTerragruntProvider } from './providers';
 import { ExtractionResult, TerragruntDependencyTypes } from './util';
 
-const githubRefMatchRegex = /github.com([/:])(?<project>[^/]+\/[a-z0-9-.]+).*\?ref=(?<tag>.*)$/;
+const githubRefMatchRegex = /github.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/;
```
Comment (viceice) [inline lib/manager/terragrunt/modules.ts:10]:
```suggestion
const githubRefMatchRegex = /github\.com([/:])(?<project>[^/]+\/[a-z0-9-_.]+).*\?ref=(?<tag>.*)$/i;
```

**Review by viceice — COMMENTED** [review]
Should have new tests.

**Comment by raspy8766** [comment]
> Should have new tests.

@viceice Added some unit tests for testing the github source regex

**Review by viceice — COMMENTED** [review]
My expectation was to add tests for the function which are suing the regexes and not the regex directly 🤔 

@rarkins  Should we ok with only testing the regex directly?

**Comment by rarkins** [comment]
I think that's ok, thanks

**Review by viceice — APPROVED** [review]

**Pull request merged**